src/Streamly/Internal/Unicode/Stream.hs

```diff
@@ -145,13 +145,17 @@ encodeLatin1' = S.map convert
     convert c =
         let codepoint = ord c
         in if codepoint > 255
-           then error $ "Streamly.String.encodeLatin1 invalid " ++
+           then error $ "Streamly.Unicode.encodeLatin1 invalid " ++
                       "input char codepoint " ++ show codepoint
            else fromIntegral codepoint
 
--- | Like 'encodeLatin1'' but silently truncates and maps input characters beyond
--- 255 to (incorrect) chars in 0-255 range. No error or exception is thrown
--- when such truncation occurs.
+-- XXX Should we instead replace the invalid chars by NUL or whitespace or some
+-- other control char? That may affect the perf a bit but may be a better
+-- behavior.
```

Comment:
We can raise an issue about this. Someone will eventually want this. We can change it then.

Issues are more easily trackable than `XXX` comments. We should come up with a process to track all `XXX` comments in the code and order them in the descending order of the date.

Reply:
yes. writing a comment is more lightweight though. extracting all of them and displaying module wise would be nice. Then we can work on removing them.